bin/lock_dev_tool.ml

```diff
@@ -72,11 +72,17 @@ let solve ~dev_tool ~local_packages =
     |> Dune_pkg.Sys_poll.solver_env_from_current_system
     |> Memo.of_reproducible_fiber
     >>| Option.some
-  and* workspace = Workspace.workspace () in
+  and* workspace =
+    let+ workspace = Workspace.workspace () in
+    if (Lazy.force Bin_dev_tools.is_enabled) then
```

Comment:
What is the purpose of `Bin_dev_tools.in_enabled` here? It seems like an unnecessary indirection when you can query `Dune_rules.Compile_time.bin_dev_tools` instead.

Reply:
Indeed, I didn't realize it when we took a look at it, but the pattern matching can directly happen here.